### PR TITLE
Kofwana's branch   rnn

### LINKED ISSUE #2: Create RNN model

### PR DESCRIPTION
**Pull request by Kofwana**
This Pull Request addresses critical issues in the sentiment analysis model, introducing a more stable padding approach, refining code structure, and implementing progress bars for transparent training and evaluation tracking. Achieving a validation accuracy of 76.62% after 5 epochs, the PR also highlights dataset labeling limitations (as Noah has mentioned). Furthermore, the upcoming addition of visualizations, including training vs. validation loss plots and example predictions, will offer a comprehensive overview of the model's performance. Please give any feedback you think is necessary for me to consider.

### CODE REVIEW

**Review by afq5024 — APPROVED** [review]

**Comment by NoahSchiro** [comment]
The branch has merge conflicts. There's a little box that states this and has a button that says resolve conflicts. We can't merge until those conflicts are resolved

File: src/utils.py [hunk 1]
```diff
@@ -1,7 +1,37 @@
...
+
+    # Label encoding for target
+    label_encoder = LabelEncoder() 
+    df['target'] = label_encoder.fit_transform(df['target'])
+
+    # Convert to a list of tuples
+    data = list(zip(df["target"].tolist(), df["text"].tolist()))
+
+    # Shuffle and sample the data
+    data = sample(data, int(len(data) * sample_fraction))
```
Comment (NoahSchiro) [inline src/utils.py:23]: This line here means we are only taking some fraction of the dataset. We want to use the whole dataset ideally.

**Comment by NoahSchiro** [comment]
The system appears to work on my machine. Good work Kofwana this project will be an easy A for us.

For us to merge, we need that src/utils.py merge conflict to be resolved. Also the way that you have changed the data preprocessing has broken my stuff. Don't worry about it, I'll fix it in my PR.

**Review by azizakaa — APPROVED** [review]

File: src/utils.py (hunk 1 above)
Comment (Kofwana) [reply src/utils.py:23]: Now, the function will use the entire dataset without sampling. I modified the get_data_torchtext function to remove the sampling step.
